### ISSUE TITLE
Use ISO-8601 format for CrossRef timestamp

### ISSUE THREAD

**Issue by rgmz** [spdx/LicenseListPublisher#124]
The current [CrossRef timestamp format is `yyyy-MM-dd - HH:mm:ss`](https://github.com/spdx/LicenseListPublisher/blob/2b565fb38a3d9b23a8bb28f5d39585f943e1ce6a/src/org/spdx/crossref/Timestamp.java#L37), which requires a custom formatter to parse and has an ambiguous timezone (I sought out this repository to double-check whether it was UTC):

```sh
# This example is using Java, but the challenge is the same regardless of language.
jshell> import java.time.ZonedDateTime;
jshell> ZonedDateTime.parse("2021-10-28 - 17:35:32");
|  Exception java.time.format.DateTimeParseException: Text '2021-10-28 - 17:35:32' could not be parsed at index 10
...
jshell> import java.time.format.DateTimeFormatter;
jshell> ZonedDateTime.parse("2021-10-28 - 17:35:32", DateTimeFormatter.ofPattern("uuuu-MM-dd - HH:mm:ss");
| Exception java.time.format.DateTimeParseException: Text '2021-10-28 - 17:35:32' could not be parsed: ... Unable to obtain ZoneId from TemporalAccessor...
```

While this — admittedly — is a very minor issue, it would be nice for the license-list-data to use ISO-8601 timestamps, which are unambiguous and simple to parse (e.g. `ZonedDateTime.parse("2021-10-28T17:35:32Z")`).

**Comment by goneall** [comment]
@rgmz Completely agree with you on this issue.  Thanks for pointing this out.

There is a [standard SPDX time/date format](https://github.com/spdx/Spdx-Java-Library/blob/42fcd5bc77ee49afc1bfb2de992679162bbcb3cd/src/main/java/org/spdx/library/SpdxConstants.java#L338): `yyyy-MM-dd'T'HH:mm:ss'Z'` which should be used - looks like the same as you are recommending.

I don't think it would break anything to switch formats over to the standard - I'll create a PR.

**Comment by goneall** [comment]
Resolved with PR #125
